test/xml_parser.ts

```diff
@@ -21,6 +21,16 @@ module.exports.parse = function (assert: any) {
     assert.done();
 };
 
+module.exports.parse_with_flags = function (assert: any) {
+    const filename = __dirname + "/../../test/fixtures/parser.xml";
+    const str = fs.readFileSync(filename, "utf8").replace(/[\r]+/g, '');
+
+    const doc = libxml.parseXml(str, {replaceEntities: true, validateEntities: true});
+    // @ts-ignore
+    assert.equal(18, doc._ref.parseFlags);
```

Comment:
We shouldn't access` _ref` outside the core lib files.

Implement and call Document.getParseFlags() instead.